core/state/state_object.go

```diff
@@ -79,7 +80,9 @@ type StateObject struct {
 	trie Trie // storage trie, which becomes non-nil on first access
 	code Code // contract bytecode, which gets set when code is loaded
 
-	originStorage  Storage // Storage cache of original entries to dedup rewrites, reset for every transaction
+	sharedOriginMap *sync.Map // Storage cache of original entries to dedup rewrites, reset for every transaction
```

Comment:
`sharedOriginMap`, the `Map` is a bit duplicated for me since it is already defined as `sync.Map`.  `sharedOriginStorage` seems better.